### PR TITLE
doc: update lld-flavor ref

### PR DESCRIPTION
**Pull request by cuishuang**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
r? rust-lang/compiler

**Comment by fee1-dead** [comment]
r? compiler

File: src/doc/rustc/src/codegen-options/index.md [hunk 1]
```diff
@@ -263,7 +263,7 @@ flavor. Valid options are:
 * `lld-link`: use the LLVM `lld` executable with the [`-flavor link`
   flag][lld-flavor] for Microsoft's `link.exe`.
 
-[lld-flavor]: https://lld.llvm.org/Driver.html
+[lld-flavor]: https://releases.llvm.org/12.0.0/tools/lld/docs/Driver.html
```
Comment (klensy) [inline src/doc/rustc/src/codegen-options/index.md:266]: Why 12? And looks like this description for `Mach-O` version of lld. Can't find description for flavors on llvm site.

**Review by wesleywiser — APPROVED** [review]
Thanks @cuishuang! I looked briefly to see if there was a non-MachO specific page detailing linker flavors we could link to but couldn't find one. Since the page is the same one we had before, just with a fixed url, I don't see any reason to hold this change up on that.

**Pull request merged**